### PR TITLE
Game state + window detection

### PR DESCRIPTION
**Pull request by rummanwaqar**

### CODE REVIEW

File: damage_window.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        '''
+        self.x = x
+        self.y = y
+        self.w = w
+        self.h = h
+
+    def corners(self):
+        '''
+        returns top-left and bottom-right corners
+        '''
```
Comment (megalphian) [inline damage_window.py:17]: Bit of a misleading function name

File: damage_window.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        self.levelStars.draw(img)
+
+
+class DamageWindow(object):
+    def __init__(self, anchor, dim, padding, img=None):
+        self.damage_bars = DamageWindow.GenerateDamageBars(anchor, dim, padding, 10)
+
+        # this section is used to test if window is open
+        test_dim = (0.07 * dim[0], (dim[1] + padding) * 3 - padding)
+        self.open_test_section = BoundingBoxObject(anchor[0] - test_dim[0] - 2, anchor[1], test_dim[0], test_dim[1], (255,255,0))
```
Comment (megalphian) [inline damage_window.py:86]: A couple of magic numbers. Is this test code necessary?

File: damage_window.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    
+    # window width (used to make sure the window is fully expanded)
+    WINDOW_WIDTH = 0
+
+    @staticmethod
+    def InitFromCoods(top_left = [1777, 326], width = 128, height = 36, padding = 9):
+        return DamageWindow(top_left, (width, height), padding)
+
+    @staticmethod
+    def InitFromFrame(img, wait_expansion = False):
```
Comment (megalphian) [inline damage_window.py:131]: Kinda seems like there are 3 methods in here which seem to use the same img argument. Fair to make this a non static class with that shared argument?

File: game_manager.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    def draw(self, img):
+        if not self.initialized:
+            cv2.putText(img, 'NOT INITIALIZED', (50,50), cv2.FONT_HERSHEY_PLAIN, 1.2, (0, 0, 255), 2)
+        elif not self.window_open:
+            cv2.putText(img, 'WINDOW CLOSED', (50,50), cv2.FONT_HERSHEY_PLAIN, 1.2, (0, 0, 255), 2)
+        else:
+            self.damage_window.draw(img)
+            stage_output = str(self.stages[-1])
+            cv2.putText(img, stage_output, (50,50), cv2.FONT_HERSHEY_PLAIN, 1.2, (0, 0, 255), 2)
```
Comment (megalphian) [inline game_manager.py:118]: Nitpicky, but you could DRY the putText function calls with a private method

**Review by megalphian — COMMENTED** [review]
Everything looks good. Lmk what you think of the comments and if you're thinking of fixing them

File: damage_window.py (hunk 2 above)
Comment (rummanwaqar) [reply damage_window.py:86]: Yes it is used to test if the window is open or not

File: damage_window.py (hunk 1 above)
Comment (rummanwaqar) [reply damage_window.py:17]: returns corners of a rectangle. you got a suggestion

File: damage_window.py (hunk 3 above)
Comment (rummanwaqar) [reply damage_window.py:131]: Ya that could be done but for now seems like too much work for no real benefit

File: game_manager.py (hunk 4 above)
Comment (rummanwaqar) [reply game_manager.py:118]: for now gonna merge it this way but will keep in mind during next iteration

**Comment by rummanwaqar** [comment]
Thanks for the review. Will try to incorporate your suggestions in the next iteration. For now wanna just move forward

**Pull request merged**